### PR TITLE
Skip DNS lookup that is not used

### PR DESCRIPTION
**Pull request by bratseth**
Look up the local address on demand only.
Since this field was (errounously) made protected,
this is strictly an API change, however I could not find any use
of the field in any known repo using Vespa.

cc @baldersheim

### CODE REVIEW

**Review by baldersheim — APPROVED** [review]
👍

**Comment by baldersheim** [comment]
The pragmatic approach does make life easier to handle.

File: jdisc_http_service/src/main/java/com/yahoo/jdisc/http/filter/DiscFilterRequest.java [hunk 1]
```diff
@@ -100,12 +91,17 @@ public void setRemoteAddr(String remoteIpAddress) {
...
     public String getLocalAddr() {
-        if (null == localAddress.getAddress()) {
-            return null;
-        }
+        InetSocketAddress localAddress = localAddress();
+        if (localAddress.getAddress() == null) return null;
         return localAddress.getAddress().getHostAddress();
     }
 
+    protected InetSocketAddress localAddress() {
```
Comment (bjorncs) [inline jdisc_http_service/src/main/java/com/yahoo/jdisc/http/filter/DiscFilterRequest.java]: I had prefer making this deprecated or removing it altogether. Any future users of this method may not know about the implicit DNS lookup. The method looks like a getter but may actually do IO, which is bad design.

**Review by bjorncs — APPROVED** [review]

File: jdisc_http_service/src/main/java/com/yahoo/jdisc/http/filter/DiscFilterRequest.java (hunk 1 above)
Comment (bratseth) [reply jdisc_http_service/src/main/java/com/yahoo/jdisc/http/filter/DiscFilterRequest.java]: My reasoning was that the field was protected for a reason, but ok.

**Pull request merged**